### PR TITLE
Visuals

### PR DESCRIPTION
**Pull request by grezisek1**

### CODE REVIEW

File: data.js [hunk 1]
```diff
@@ -6,40 +6,96 @@ export const SCENE_HEIGHT = 1080;
...
+    {
+        name: "accel",
+        velocity: 100,
+        velocityDamp: 1.01,
+        reloadTime: 3.2,
+        despawnTime: 32,
+        damage: 12,
         size: 128,
     },
 ];
```
Comment (NerdbordMentor) [inline data.js:101]: **suggestion**: Consider using a more descriptive name for `gametimeScale` to clarify its purpose.

File: gameplay-controller.js [hunk 2]
```diff
@@ -35,11 +33,13 @@ export default class GameplayController {
             kills: 0,
             milestone: 0,
             weapons: [0],
+            score: 0,
         };
```
Comment (NerdbordMentor) [inline gameplay-controller.js:37]: **issue**: The use of a hardcoded random value (`const rand = 1;`) may not provide the intended randomness. Consider using `Math.random()` for true randomness.

File: gameplay-controller.js [hunk 3]
```diff
@@ -81,54 +96,77 @@ export default class GameplayController {
...
+        const dir = [Math.random() - 0.5, Math.random() - 0.5];
+        const mSq = dir[0]**2 + dir[1]**2;
+        let x;
+        let y;
+        if (mSq < 0.001) {
+            x = spawnRadius;
+            y = 0;
+        } else {
+            const m = mSq**0.5;
+            x = dir[0] * spawnRadius / m;
```
Comment (NerdbordMentor) [inline gameplay-controller.js:112]: **praise**: Excellent implementation of highscore tracking, which enhances replayability.

File: survive-the-monsters.js [hunk 4]
```diff
@@ -30,9 +30,11 @@ const scenes = {
 scenes.main.preload = function() {
     this.load.image("player", "sprites/players/mvp_player/idle_0.png");
```
Comment (NerdbordMentor) [inline survive-the-monsters.js:31]: **issue**: The `workaroundSpawnGlitch` function is added to handle visibility issues. Verify that this does not introduce any new bugs or side effects.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Overall Feedback

The code changes in this pull request introduce significant enhancements, including animations, improved sprites, and new gameplay elements like particle effects and explosions. The code is well-structured and meets the task requirements, enhancing the game's visual and interactive aspects for a more engaging user experience. However, some areas require further attention to ensure optimal performance and maintainability.

### Things Missing:

- [] Ensure that all new features are thoroughly tested to prevent potential bugs.
- [] Verify that the `workaroundSpawnGlitch` function is necessary and does not introduce unintended side effects.

### Things That Went Well:

- The addition of animations and improved sprites significantly enhances the visual appeal of the game.
- The introduction of a scoring system and highscore tracking adds a competitive element that can increase player engagement.
- The refactoring of enemy and bullet spawning logic improves code readability and maintainability.
- The refactoring of sprite loading logic is clean and improves maintainability.
- The use of dynamic speed assignment for enemies based on type is a good improvement for gameplay dynamics.

**Decision: REQUEST_CHANGES**

The decision to request changes is based on the need to ensure that all new features are thoroughly tested and that the `workaroundSpawnGlitch` function does not introduce unintended side effects. Addressing these issues will ensure the code functions as intended and maintains high quality.

**Comment by grezisek1** [comment]
thanks

File: gameplay-controller.js [hunk 5]
```diff
@@ -35,11 +33,13 @@ export default class GameplayController {
             kills: 0,
             milestone: 0,
             weapons: [0],
+            score: 0,
         };
         this.controllerState = {
```
Comment (NerdbordMentor) [inline gameplay-controller.js:38]: **praise**: Good use of `Math.random()` for generating randomness, which improves the unpredictability of weapon spawning.

File: survive-the-monsters.js (hunk 4 above)
Comment (NerdbordMentor) [inline survive-the-monsters.js:31]: **issue**: Ensure that the `workaroundSpawnGlitch` function does not introduce any new bugs or side effects. Consider adding tests to verify its behavior.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Overall Feedback

The code changes in this pull request introduce several enhancements, including improved sprites, animations, and new gameplay elements like particle effects and explosions. The code is generally well-structured and meets the task requirements, enhancing the game's visual and interactive aspects for a more engaging user experience. However, there are a few areas that require further attention to ensure optimal performance and maintainability. The previous review's concerns about testing new features and verifying the necessity of the `workaroundSpawnGlitch` function remain relevant and should be addressed to ensure the code functions as intended.

### Things Missing:

- [] Ensure that all new features are thoroughly tested to prevent potential bugs.
- [] Verify that the `workaroundSpawnGlitch` function is necessary and does not introduce unintended side effects.

### Things That Went Well:

- The addition of animations and improved sprites significantly enhances the visual appeal of the game.
- The introduction of a scoring system and highscore tracking adds a competitive element that can increase player engagement.
- The refactoring of enemy and bullet spawning logic improves code readability and maintainability.
- The refactoring of sprite loading logic is clean and improves maintainability.
- The use of dynamic speed assignment for enemies based on type is a good improvement for gameplay dynamics.

File: survive-the-monsters.js (hunk 4 above)
Comment (grezisek1) [reply survive-the-monsters.js:31]: this is only 1 frame of sprite visible at top left corner. workaroud works - i can tell using eyes

**Review by NerdbordMentor — APPROVED** [review]
Force approved. Hope you're not cheating!

**Pull request merged**